sqlx-mysql/src/testing/mod.rs

```diff
@@ -117,42 +139,27 @@ async fn test_context(args: &TestArgs) -> Result<TestContext<MySql>, Error> {
     conn.execute(
         r#"
         create table if not exists _sqlx_test_databases (
-            db_id bigint unsigned primary key auto_increment,
+            db_name text primary key,
             test_path text not null,
             created_at timestamp not null default current_timestamp
         );
     "#,
     )
     .await?;
 
-    // Record the current time _before_ we acquire the `DO_CLEANUP` permit. This
-    // prevents the first test thread from accidentally deleting new test dbs
-    // created by other test threads if we're a bit slow.
-    let now = SystemTime::now()
-        .duration_since(SystemTime::UNIX_EPOCH)
-        .unwrap();
-
-    // Only run cleanup if the test binary just started.
-    if DO_CLEANUP.swap(false, Ordering::SeqCst) {
-        do_cleanup(&mut conn, now).await?;
-    }
+    let db_name = MySql::db_name(args);
+    do_cleanup(&mut conn, &db_name).await?;
 
-    query("insert into _sqlx_test_databases(test_path) values (?)")
+    query("insert into _sqlx_test_databases(test_path) values ($1, $2)")
```

Comment:
This isn't valid syntax in MySQL.

Reply:
You're totally right that it isn't the right syntax, also didn't name the column :-1: .

For the general tests I can't run them because I get this error:
`> ./x.py`
```
 # unit test core
 $ cargo test --no-default-features --manifest-path sqlx-core/Cargo.toml --features json,offline,migrate,_rt-async-std,_tls-rustls 
   Compiling sqlx-core v0.8.1 (/home/andreasliljeqvist/git/sqlx/sqlx-core)
error[E0425]: cannot find value `provider` in this scope
  --> sqlx-core/src/net/tls/tls_rustls.rs:97:54
   |
97 |     let config = ClientConfig::builder_with_provider(provider.clone())
   |                                                      ^^^^^^^^ not found in this scope
```

I pushed another commit that I hope will fix your comment.

Reply:
I really need to just get rid of `x.py`. I don't use it and it's not tested in CI so there's no reason for it to get fixed or kept up to date unless someone else bothers to.

Reply:
@abonander , sorry to tag, but wanted to ask: will you have time to see the corrected changes?

Reply:
I'm currently on vacation and don't have access to a computer to cut a release. I will set aside time for SQLx next week.

Reply:
thanks for the quick reply. have a great vacation!

Reply:
I tried to integrate the changes as we have a lot of tests (https://github.com/kamu-data/kamu-cli/pull/919) and many of them use `sqlx` + `nextest` combination.

In the case of PostgreSQL, everything works as expected, but in the case of MySQL, the changes are not finalized:
```
failed to connect to setup test database: Database(MySqlDatabaseError { code: Some("42000"), number: 1064, message: "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '\"_sqlx_test_b4OPm0XBsXli4d_oBPTB7Qy_7snq3Wiwey-n0eGShY0FeWK3ecaW\"' at line 1" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Based on the PostgreSQL implementation, I finished MySQL one: https://github.com/bonega/sqlx/pull/1 -- @bonega , please take a look and if you wish, you can use my changes (paste into your branch).

---

We have a large number of tests (`sqlx` + `nextest`), so in general I think it's fair to call it successful change testing

Reply:
Addendum: during testing, a regression on not deleting SQLite databases was identified -- it is also fixed in https://github.com/bonega/sqlx/pull/1